src/components/MarkdownWrapper/MarkdownWrapper.style.tsx

```diff
@@ -51,9 +52,18 @@ export const MarkdownWrapperStyled = styled.div.attrs<MarkdownWrapperStyledProps
   }
 `;
 
-export const ButtonLink = styled.a.attrs((props) => ({
+type ExtraEventType = { trackEvent?: string };
+export const ButtonLink = styled.a.attrs((props: Record<string, any>) => ({
   ...props,
   className: `fr-btn ${props.className || ''}`,
+  ...((props as ExtraEventType)?.trackEvent && {
```

Comment:
weird to have a deconstructor here

Reply:
I can't find it but if you prefer we can use an intermediate variable (personally, I find it heavier).